src/compiler/checker.ts

```diff
@@ -19222,6 +19222,12 @@ namespace ts {
         }
 
         function isValidSpreadType(type: Type): boolean {
+            if (type.flags & TypeFlags.TypeVariable) {

```

Comment:
would it make sense to use `TypeFlags.Instantiable` and `getBaseConstraintOfType` instead?
IIUC that would also handle deferred conditional types

Reply:
Yeah. Let me update this in a few. Thanks again.

Reply:
Updated!